src/test/java/com/fasterxml/jackson/dataformat/xml/lists/StringListRoundtripTest.java

```diff
@@ -0,0 +1,170 @@
+package com.fasterxml.jackson.dataformat.xml.lists;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+
+import static com.fasterxml.jackson.dataformat.xml.deser.FromXmlParser.Feature.PROCESS_XSI_NIL;
+import static com.fasterxml.jackson.dataformat.xml.ser.ToXmlGenerator.Feature.WRITE_NULLS_AS_XSI_NIL;
+import static java.util.Arrays.asList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+// [dataformat-xml#584]
+public class StringListRoundtripTest {
```

Comment:
Looks good, but one big(ger) ask: in addition to root values (which are actually harder to support, bit less common to use), would it be possible to also add "as-property" cases where String arrays, Collections, Maps are properties of POJOs? That would we can double-check that empty Strings do not accidentally become `null`s, which is my main concern.